geo_to_hca/geo_to_hca.py

```diff
@@ -24,16 +24,15 @@ def fetch_srp_accession(geo_accession: str) -> str:
     Function to retrieve an SRA study accession given a GEO accession.
     """
     srp = sra_utils.SraUtils.get_srp_accession_from_geo(geo_accession)
-    if not srp.empty:
+    if srp and not srp.empty:
         if srp.shape[0] == 1:
             srp = srp.iloc[0]["study_accession"]
         elif srp.shape[0] > 1:
-            raise Exception("More than 1 accession has been found. Please enter re-try with a single SRA Study "
+            raise IndexError("More than 1 accession has been found. Please enter re-try with a single SRA Study "
                             "accession.")
     else:
-        raise Exception(f"Could not recognise GEO accession {geo_accession}; is it a GEO Superseries? If yes, please "
-                        f"re-try with a "
-                        "subseries accession")
+        raise IndexError(f"Could not find SRA accession for GEO accession {geo_accession}; is it a GEO super-series? "
```

Comment:
What about `KeyError` instead of `IndexError`?

```suggestion
        raise KeyError(f"Could not find SRA accession for GEO accession {geo_accession}; is it a GEO super-series? "
```

Reply:
I think IndexError is better. It's a list and it's not empty